DiamondLightSource/python-artemis
# 399 random order tests

Fixes #399 
Randomises the order in which tests are run, including in CI, in order to help identify any interdependent tests.

Tests will fail currently; they are fixed in #405, which is merged into main now

### To test:
0. You might need to `pip install pytest-random-order` (or run `./dls_dev_env.sh` again)
1. Run `pytest --random-order -m "not s03` 
2. Observe that some ispyb tests fail sometimes
3. Merge main
4. Rerun tests a few times and see they always pass
